lib/python/ray/worker.py

```diff
@@ -775,6 +775,17 @@ def put(value, worker=global_worker):
   worker.put_object(objectid, value)
   return objectid
 
+def select(objectids, num_objects=0, worker=global_worker):
+  check_connected(worker)
+  ready_ids = raylib.is_ready(worker.handle, objectids)
+  while sum(ready_ids) < num_objects:
+    ready_ids = raylib.is_ready(worker.handle, objectids)
+  result = [index for index, _ in enumerate(objectids) if ready_ids[index]] 
+  if (num_objects != 0):
+     return result[0:num_objects]
+  else: return result
```

Comment:
This method shouldn't loop. It should send a message to the scheduler that includes `num_objects`, and the scheduler shouldn't return until that number of objects is ready.